### PR TITLE
ENH: Allow convert_dtypes to convert to pd.ArrowDtype

### LINKED ISSUE #49997: ENH: Allow `convert_dtypes` to convert to `pd.ArrowDtype`
### Feature Type

- [x] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

Currently `df.convert_dtypes()` will convert to nullable types (Int64, Boolean, etc), but it would be nice if there was an option to opt into the pyarrow-backed nullable types via `pd.ArrowDtype`

### Feature Description

Maybe either

* New keyword argument `nullable_type="pandas"|"pyarrow"`
* Refactor `io.nullable_backend` global setting to `mode.nullable_backend` and allow that global config to apply to `df.convert_dtypes`

### Alternative Solutions

For loop over the existing DataFrame object using `astype` and constructing `pd.ArrowDtype`?

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #49997 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Additionally changed `io.nullable_backend` to `mode.nullable_backend` to make it more general

### CODE REVIEW

File: pandas/tests/frame/methods/test_convert_dtypes.py [hunk 1]
```diff
@@ -41,3 +43,56 @@ def test_convert_dtypes_retain_column_names(self):
...
+            }
+        )
+        tm.assert_frame_equal(result, expected)
+
+    def test_pyarrow_nullable_backend_already_pyarrow(self):
+        pytest.importorskip("pyarrow")
+        expected = pd.DataFrame([1, 2, 3], dtype="int64[pyarrow]")
+        with pd.option_context("mode.nullable_backend", "pyarrow"):
+            result = expected.convert_dtypes()
+        tm.assert_frame_equal(result, expected)
```
Comment (jrbourbeau) [inline pandas/tests/frame/methods/test_convert_dtypes.py:98]: Does setting `mode.nullable_backend` to `"pyarrow"` work when the original dtypes are already numpy-backed extension dtypes? For example, something like

```python
        expected = pd.DataFrame([1, 2, 3], dtype="Int64")
        with pd.option_context("mode.nullable_backend", "pyarrow"):
            result = expected.convert_dtypes()
        tm.assert_frame_equal(result, expected)
```

or adding a `df.convert_dtypes()` line before `with pd.option_context("mode.nullable_backend", "pyarrow"):` in `test_pyarrow_nullable_backend`

File: pandas/core/dtypes/cast.py [hunk 2]
```diff
@@ -979,6 +983,11 @@ def convert_dtypes(
     infer_objects : bool, defaults False
         Whether to also infer objects to float/int if possible. Is only hit if the
         object array contains pd.NA.
+    nullable_backend : str, default "pandas"
```
Comment (jrbourbeau) [inline pandas/core/dtypes/cast.py:986]: How does this interact with the `mode.nullable_backend` option? My guess is folks would expect `nullable_backend` to take precedent if it's specified, but for `mode.nullable_backend` to take precedent if `nullable_backend` isn't given. 

Here's a tiny example of the use case where I'm asking what happens:

```python
with pd.option_context("mode.nullable_backend", "pyarrow"):
    result = expected.convert_dtypes(nullable_backend="pandas")
```

**Review by jrbourbeau — COMMENTED** [review]
Nice @mroeschke! 

I definitely think the `io.nullable_backend` -> `mode.nullable_backend` change makes sense given it's being generalized outside of just `read_*` I/O functions.

File: pandas/tests/frame/methods/test_convert_dtypes.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/frame/methods/test_convert_dtypes.py:98]: Yup this should convert the numpy-backed types to pyarrow-backed types. Will add a unit test for it

File: pandas/core/dtypes/cast.py (hunk 2 above)
Comment (mroeschke) [reply pandas/core/dtypes/cast.py:986]: This line specifically is a private helper function that backs the public `Series/DataFrame.convert_dtypes`.

The public `Series/DataFrame.convert_dtypes` will not accept a keyword argument `nullable_backend`

**Review by phofl — APPROVED** [review]
lgtm

**Pull request merged**